### PR TITLE
Tdd content

### PR DESCRIPTION
**Pull request by JBonifay**
I open this draft MR to collect feedback, if anyone wants to work on this page with me don't hesitate !

### CODE REVIEW

File: test-driven-development.md [hunk 1]
```diff
@@ -0,0 +1,171 @@
+# Test driven development
+
+## TDD rules
+
+* You're not allowed to write any production code until you've got a unit test failing first
+
+> Warning: **not compiling is failing !!!**
+
+If a test is fails to compile, stop writing the test and write production code to make the test compile.
```
Comment (eriklautenschlaeger) [inline test-driven-development.md:9]: -is

**Comment by eriklautenschlaeger** [comment]
There are a number of typos etc. but as far as I see it this is not meant for being published but only in a draft state? 

//Edit: NVM, I should take a reading class :-D

**Comment by JBonifay** [comment]
Hi Erik ! 
Haha no problem, yes I did put it as a draft, it will need for sure a correction for the English anyway, I'm pretty sure someone can rephrase and improve all after.

I put it as draft to allow people to use it and improve it while I'm adding content, by the way if you wish don't hesitate!

**Comment by valentinajemuovic** [comment]
I see the message: "This pull request is still a work in progress. Draft pull requests cannot be merged."

Regarding spelling/grammar mistakes:
1. I make a lot of spelling/grammar mistakes, esp. when I'm writing fast. 
2. For that reason, I ended up getting Grammarly which I use as proofreading for my LinkedIn articles & commercial work too. Therefore, for any text here on Tech Excellence, when I go into edit mode, I can make corrections too with Grammarly.
3. Since there'll be "multiple eyes" on this knowledge base, so it means at some point it would be caught.

Indeed, this was one personal challenge I faced for many years:
- For several years, I had this problem: anything that I would write, it was never good enough. Every time I reviewed my work, I wasn't happy with what I wrote. This blocked me from publishing anything for about 10 years.
- During the past year (due to LinkedIn posts), I changed my perspective - I see it all as continuous drafts, and I get feedback from viewers.

So from my perspective, it's fine to do the merge...

**Comment by JBonifay** [comment]
Good idea, this will perhaps this will attract others to contribute.

**Comment by valentinajemuovic** [comment]
@JBonifay approved. Since I see you wrote an extensive page even for a first draft, I added you now in the maintainers team for this repo which means you can directly commit to the main branch (i.e. TBD approach).

I will also be doing some corrections/updates to the page, since TDD is of key interest here.

**Pull request merged**